File: app/src/main/java/com/ninhttd/devtest/data/remote/interceptor/NetworkInterceptor.java
package com.ninhttd.devtest.data.remote.interceptor;

import com.ninhttd.devtest.data.remote.exception.NetworkException;
import com.ninhttd.devtest.utils.NetworkUtils;

import java.io.IOException;

import okhttp3.Interceptor;
import okhttp3.Response;

public class NetworkInterceptor implements Interceptor {
    @Override
    public Response intercept(Chain chain) throws IOException {
        if(!NetworkUtils.isNetworkAvailable()){
            throw new NetworkException();
        }
        return chain.proceed(chain.request());
    }
}

File: app/src/main/java/com/ninhttd/devtest/data/remote/interceptor/NetworkInterceptor.kt
package com.ninhttd.devtest.data.remote.interceptor

import com.ninhttd.devtest.data.remote.exception.NetworkException
import com.ninhttd.devtest.utils.NetworkUtils

import java.io.IOException

import okhttp3.Interceptor
import okhttp3.Response

class NetworkInterceptor : Interceptor {
    @Throws(IOException::class)
    override fun intercept(chain: Interceptor.Chain): Response {
        if (!NetworkUtils.isNetworkAvailable) {
            throw NetworkException()
        }
        return chain.proceed(chain.request())
    }
}
